hera_cal/lst_stack/metrics.py

```diff
@@ -118,63 +118,93 @@ def get_nightly_predicted_variance(
 
     auto = auto_stats.mean[(bl[0], bl[0], bl[2])] * auto_stats.mean[(bl[1], bl[1], bl[2])]
 
-    per_day_expected_var = [None] * len(stacks)
-    dtdf = dt * df
+    dtdf = stack.dt * stack.df
 
-    for i, stack in enumerate(stacks):
-        gf = stack.get_flags(bl)
-        per_day_expected_var[i] = np.abs(auto[i] / dtdf / stack.get_nsamples(bl))
-        per_day_expected_var[i][gf] = np.inf
+    gf = stack.get_flags(bl)
+    per_day_expected_var = np.abs(auto / dtdf / stack.get_nsamples(bl))
+    per_day_expected_var[gf] = np.inf
 
     return per_day_expected_var
 
 
-def get_zscores(
+def get_squared_zscores(
     auto_stats: LSTBinStats,
     cross_stats: LSTBinStats,
-    dt: float,
-    df: float,
-    stacks: list[UVData],
+    stack: LSTStack,
     central: str = 'mean',
     std: str = 'autos'
-) -> list[UVData]:
+) -> list[UVFlag]:
     """
-    Obtain Z-scores as a list of UVData objects.
+    Obtain squared Z-scores as a list of UVFlag objects in metrics mode.
+    Parameters
+    ----------
+    auto_stats : LSTBinStats
+        LST-binned statistics for the autos. Used to predict the variance of the
+        cross-correlations if std=='autos'.
+    cross_stats : LSTBinStats
+        LST-binned statistics for the cross-correlations. Used to compute the Z-scores,
+        by obtaining the mean or median.
+    stack : LSTStack
+        The LST-stack of data over nights, as an LSTStack object (which is essentially
+        a UVData object).
+    central : {'mean', 'median'}
+        The central value to use for the Z-score. If 'mean', the mean of the
+        visibilities over nights is used. If 'median', the median is used.
+    std : {'autos', 'std', 'mad'}
+        The standard deviation to use for the Z-score. If 'autos', the autos are used
+        to predict the variance of the cross-correlations. If 'std' or 'mad', the
+        cross-correlations are used, and the statistic is estimated over nights from
+        the sample.
     """
-    zstacks = [stack.copy() for stack in stacks]  # UVData objects
+    zstack = UVFlag(stack, mode='metric', label='zsquare')  # [UVFlag(stack, mode='metric', label='zsquare') for stack in stacks]
```

Comment:
is there a reason this comment is still here?

Reply:
no, I removed it now.